### PR TITLE
[SYCL] Add buffer dimensions restriction

### PR DESCRIPTION
**Pull request by MochalovaAn**
According to the specification, a buffer can have dimensions of only 1, 2, and 3. In the old version, "static_assert" was used to limit it, which did not throw an exception when creating a pointer to a buffer with an incorrect dimension. In this patch it is impossible to create not only an object of the buffer class with incorrect dimensions, but also a link to it.
Also, constructor calls in the accessos class that tried to instantiate a buffer with a zero dimension were fixed.

Signed-off-by: amochalo <anastasiya.mochalova@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
Please, apply clang-format.

File: sycl/include/CL/sycl/accessor.hpp [hunk 1]
```diff
@@ -1250,7 +1251,7 @@ class accessor<DataT, Dimensions, AccessMode, access::target::image_array,
 };
 
 } // namespace sycl
-} // __SYCL_INLINE_NAMESPACE(cl)
+} // namespace cl
```
Comment (AlexeySachkov) [inline sycl/include/CL/sycl/accessor.hpp]: Please revert this change

File: sycl/include/CL/sycl/accessor.hpp [hunk 2]
```diff
@@ -773,18 +773,20 @@ class accessor :
...
-  template <int Dims = Dimensions, typename AllocatorT>
-  accessor(buffer<DataT, 1, AllocatorT> &BufferRef,
-           detail::enable_if_t<Dims == 0 &&
-                               (!IsPlaceH && (IsGlobalBuf || IsConstantBuf)),
-                               handler> &CommandGroupHandler)
+  template <int Dims = Dimensions, typename AllocatorT,
+   typename = typename detail::enable_if_t<
+		   (Dims == 0) && 
+                    (!IsPlaceH && (IsGlobalBuf || IsConstantBuf))>
+		    			>
```
Comment (AlexeySachkov) [inline sycl/include/CL/sycl/accessor.hpp]: Is this code properly clang-formatted?

**Comment by sergey-semenov** [comment]
Please, apply clang-format to the patch

**Comment by sergey-semenov** [comment]
The contents of the patch LGTM, but could you please fix the commit message? It seems that the description refers to the diff between this patch and its earlier version.

File: sycl/include/CL/sycl/accessor.hpp [hunk 3]
```diff
@@ -816,7 +815,7 @@ class accessor :
   template <int Dims = Dimensions, typename AllocatorT,
             typename = detail::enable_if_t<
                 (Dims > 0) && (!IsPlaceH && (IsGlobalBuf || IsConstantBuf))>>
-  accessor(buffer<DataT, Dimensions, AllocatorT> &BufferRef,
+  accessor(buffer<DataT, Dims, AllocatorT> &BufferRef,
```
Comment (romanovvlad) [inline sycl/include/CL/sycl/accessor.hpp]: Why are you changing this?

Comment (MochalovaAn) [reply sycl/include/CL/sycl/accessor.hpp]: Because calling this kind of function led to an error

Comment (MochalovaAn) [reply sycl/include/CL/sycl/accessor.hpp]: Because calling this kind of function led to an error, since this leads to instantiation of a buffer with invalid (0) parameters.
